### ISSUE TITLE
Define Grommet theme font sizes in rem & line-height in unitless numbers

### ISSUE THREAD

**Issue by goplayoutside3** [zooniverse/front-end-monorepo#6028]
## Package

lib-grommet-theme

## Is your feature request related to a problem? Please describe.

To increase platform-wide accessibility we should define our font sizes in `rem` units rather than hardcoded pixels. It's also preferred to define line-height as a unitless ratio rather than pixels.

## Describe the solution you'd like

The `size` (font-size) prop in our Grommet theme components should be redefined for paragraphs, text, and headings. Each are linked below, and also have a 'height` prop which corresponds to line-height.

Paragraphs
https://github.com/zooniverse/front-end-monorepo/blob/d9856fb4d84ea3a3c3d5bb373771cf8d095e81a2/packages/lib-grommet-theme/src/index.js#L466

Text
https://github.com/zooniverse/front-end-monorepo/blob/d9856fb4d84ea3a3c3d5bb373771cf8d095e81a2/packages/lib-grommet-theme/src/index.js#L523

Headings
https://github.com/zooniverse/front-end-monorepo/blob/d9856fb4d84ea3a3c3d5bb373771cf8d095e81a2/packages/lib-grommet-theme/src/index.js#L309

## Additional context

Building upon https://github.com/zooniverse/front-end-monorepo/pull/6007

Related to: https://github.com/zooniverse/front-end-monorepo/issues/5469
